woltka/tests/test_coverage.py

```diff
@@ -0,0 +1,126 @@
+
+        # add ranges but don't merge
+        obs = {}
+        parse_ranges(rmap, obs)
+        exp = {('S1', 'G1'): [4, (1, 100), (1, 50), (251, 300), (1, 50)],
+               ('S1', 'G2'): [1, (101, 150)],
+               ('S2', 'G2'): [3, (151, 200), (51, 100), (26, 75)],
+               ('S2', 'G3'): [2, (76, 125), (1, 50)],
+               ('S2', 'G4'): [2, (26, 75), (101, 150)],
+               ('S3', 'G1'): [4, (1, 50), (51, 100), (151, 200), (101, 150)],
+               ('S3', 'G2'): [2, (51, 100), (1, 50)]}
+        self.assertDictEqual(obs, exp)
+
+        # merge while adding
+        obs = {}
+        parse_ranges(rmap, obs, chunk=1)
+        exp = {('S1', 'G1'): [0, (1, 100), (251, 300)],
+               ('S1', 'G2'): [0, (101, 150)],
+               ('S2', 'G2'): [0, (26, 100), (151, 200)],
+               ('S2', 'G3'): [0, (1, 50), (76, 125)],
+               ('S2', 'G4'): [0, (26, 75), (101, 150)],
+               ('S3', 'G1'): [0, (1, 200)],
+               ('S3', 'G2'): [0, (1, 100)]}
+        self.assertDictEqual(obs, exp)
+
+    def test_calc_coverage(self):
+        cov = {('S1', 'G1'): [4, (1, 100), (1, 50), (251, 300), (1, 50)],
+               ('S1', 'G2'): [1, (101, 150)],
+               ('S2', 'G2'): [3, (151, 200), (51, 100), (26, 75)],
+               ('S2', 'G3'): [2, (76, 125), (1, 50)],
+               ('S2', 'G4'): [2, (26, 75), (101, 150)],
+               ('S3', 'G1'): [4, (1, 50), (51, 100), (151, 200), (101, 150)],
+               ('S3', 'G2'): [2, (51, 100), (1, 50)]}
+        obs = calc_coverage(cov)
+        exp = {'S1': {'G1': [(1, 100), (251, 300)], 'G2': [(101, 150)]},
+               'S2': {'G2': [(26, 100), (151, 200)], 'G3': [(1, 50), (
+                   76, 125)], 'G4': [(26, 75), (101, 150)]},
+               'S3': {'G1': [(1, 200)], 'G2': [(1, 100)]}}
+        self.assertDictEqual(obs, exp)
+
+    def test_write_coverage(self):
+        cov = {'S1': {'G1': [(1, 100), (251, 300)], 'G2': [(101, 150)]},
+               'S2': {'G2': [(26, 100), (151, 200)], 'G3': [(1, 50), (
+                   76, 125)], 'G4': [(26, 75), (101, 150)]},
+               'S3': {'G1': [(1, 200)], 'G2': [(1, 100)]}}
+        outdir = join(self.tmpdir, 'outdir')
+        makedirs(outdir)
+        write_coverage(cov, outdir)
+        with open(join(outdir, 'S1.cov'), 'r') as f:
+            obs = f.read().splitlines()
+        exp = ['G1\t1\t100', 'G1\t251\t300', 'G2\t101\t150']
+        self.assertListEqual(obs, exp)
+        with open(join(outdir, 'S2.cov'), 'r') as f:
+            obs = f.read().splitlines()
+        exp = ['G2\t26\t100', 'G2\t151\t200', 'G3\t1\t50',
+               'G3\t76\t125', 'G4\t26\t75', 'G4\t101\t150']
+        self.assertListEqual(obs, exp)
+        with open(join(outdir, 'S3.cov'), 'r') as f:
+            obs = f.read().splitlines()
+        exp = ['G1\t1\t200', 'G2\t1\t100']
+        self.assertListEqual(obs, exp)
+        rmtree(outdir)
```

Comment:
Might be worth a try finally to ensure the tmp directory is cleaned up when the test fails after writing coverage but asserts.

Reply:
I think that the `tearDown` method already ensures that. See: https://docs.python.org/3/library/unittest.html#unittest.TestCase.tearDown